src/Plugins/Rhetos.LightDMS/Download/DownloadHelper.cs

```diff
@@ -264,12 +264,13 @@ private async Task DownloadFromS3(FileMetadata fileMetadata, HttpContext context
                 getObjRequest.BucketName = _s3Options.BucketName;
                 if (string.IsNullOrWhiteSpace(getObjRequest.BucketName))
                     throw new FrameworkException("Missing S3 storage bucket name.");
-                
+
+                getObjRequest.Key = string.Empty;
                 var s3Folder = _s3Options.DestinationFolder;
-                if (string.IsNullOrWhiteSpace(s3Folder))
-                    throw new FrameworkException("Missing S3 folder name.");
+                if (!string.IsNullOrWhiteSpace(s3Folder))
+                    getObjRequest.Key = $"{s3Folder}/";
 
-                getObjRequest.Key = s3Folder + "/doc-" + fileMetadata.FileContentId.ToString();
+                getObjRequest.Key += $"doc-{fileMetadata.FileContentId}";

```

Comment:
same here, it might cause breaking changes to existing systems

Reply:
I reverted this file.